04/2_print_string_buffer/buffered_print.c

```diff
@@ -0,0 +1,26 @@
+#include <stdio.h>
+#include <string.h>
+
+#include "buffered_print.h"
+
+#define BUFFER_LENGTH (32)
+
+static int s_buffer_next_index = 0;
+static char s_buffer[BUFFER_LENGTH];
+
+void buffered_print(const char* src, size_t num)
+{
+    if (s_buffer_next_index == BUFFER_LENGTH) {
+        printf("The buffer is full:\n%s\n", s_buffer);
+        s_buffer_next_index = 0;
+    }
+
+    if (s_buffer_next_index == 0) {
+        strncpy(s_buffer, src, num);
+    } else {
+        strncat(s_buffer, src, num);
+    }
+
+    s_buffer[BUFFER_LENGTH - 1] = '\0';

```

Comment:
strncat correctly puts null character. so u don't need this

Reply:
u need this for only strncpy

Reply:
but if u put "12345678912345123456789123456" (when buffer_length is 16)

i think you will print "12345678912345" "23456789123456"

which is wrong.